src/tokens/str_lst/str_new.c

```diff
@@ -0,0 +1,28 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   str_new.c                                          :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: mcourtoi <mcourtoi@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/03/08 19:37:40 by mcourtoi          #+#    #+#             */
+/*   Updated: 2023/03/27 17:29:17 by mcourtoi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+t_str	*str_new(char *const str, bool is_quoted)
+{
+	t_str *const	output = malloc(sizeof(t_str));
+
+	if (!output)
+		return (NULL);
+	output->str = ft_strdup(str);
+	if (!output->str)
+		return (NULL);
```

Comment:
If this condition is true (if `ft_strdup()` fails), then you forgot to `free(output)`, and thus you have a definitely lost memory leak.

Reply:
It is true.